widget/embedded/src/store/ui.ts

```diff
@@ -8,14 +13,44 @@ interface UiState {
   currentPage: string;
   setSelectedSwap: (requestId: string | null) => void;
   setCurrentPage: (path: string) => void;
+  isActiveTab: boolean | null;
+  showActivateTabModal: boolean;
+  activateCurrentTab: (
+    swaps: PendingSwap[],
+    setCurrentTabAsActive: () => void
+  ) => void;
+  setShowActivateTabModal: (flag: boolean) => void;
 }
-
 export const useUiStore = createSelectors(
-  create<UiState>()((set) => ({
+  create<UiState>()((set, get) => ({
     connectWalletsButtonDisabled: false,
     selectedSwapRequestId: null,
     currentPage: '',
+    isActiveTab: null,
+    showActivateTabModal: false,
     setSelectedSwap: (requestId) => set({ selectedSwapRequestId: requestId }),
     setCurrentPage: (path) => set({ currentPage: path }),
+    activateCurrentTab: (swaps, setCurrentTabAsActive) => {
+      const { showActivateTabModal } = get();
+      const anyRunningSwap = swaps.some((swap) => swap.status === 'running');
+
+      if (!showActivateTabModal && anyRunningSwap) {
+        set({ showActivateTabModal: true });
+      } else if (showActivateTabModal || !anyRunningSwap) {
+        setCurrentTabAsActive();
+      }
+    },
+    setShowActivateTabModal: (flag) => {
+      set({ showActivateTabModal: flag });
+    },
   }))
 );
+
+const tabManager: TabManagerInterface = new TabManager({
+  onInit: () => useUiStore.setState({ isActiveTab: false }),
+  onClaim: () =>
+    useUiStore.setState({ isActiveTab: true, showActivateTabModal: false }),
+  onRelease: () => useUiStore.setState({ isActiveTab: false }),
```

Comment:
Is there a way to have a delay somewhere in releasing or claiming lock, so when we want to refresh an active tab in a short time, active tab doesn't change? :thinking: